src/e2e/get_liquidity_ticks.rs

```diff
@@ -0,0 +1,329 @@
+#[cfg(test)]
+pub mod e2e_tests {
+    use crate::{
+        contracts::{entrypoints::InvariantTrait, FeeTier, PoolKey, LIQUIDITY_TICK_LIMIT},
+        invariant::InvariantRef,
+        math::types::{
+            liquidity::Liquidity,
+            percentage::Percentage,
+            sqrt_price::{calculate_sqrt_price, SqrtPrice},
+        },
+    };
+    use decimal::*;
+    use ink_e2e::build_message;
+    use test_helpers::{
+        add_fee_tier, approve, create_dex, create_pool, create_position, create_tokens,
+        get_liquidity_ticks,
+    };
+    use token::{TokenRef, PSP22};
+
+    type E2EResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+    #[ink_e2e::test]
+    async fn test_get_liquidity_ticks(mut client: ink_e2e::Client<C, E>) -> E2EResult<()> {
+        let dex = create_dex!(client, InvariantRef, Percentage::from_scale(1, 2));
+        let initial_amount = 10u128.pow(10);
+        let (token_x, token_y) = create_tokens!(client, TokenRef, initial_amount, initial_amount);
+
+        let alice = ink_e2e::alice();
+
+        let fee_tier = FeeTier::new(Percentage::from_scale(1, 2), 1).unwrap();
+
+        add_fee_tier!(client, InvariantRef, dex, fee_tier, alice).unwrap();
+
+        let init_tick = 0;
+        let init_sqrt_price = calculate_sqrt_price(init_tick).unwrap();
+        create_pool!(
+            client,
+            InvariantRef,
+            dex,
+            token_x,
+            token_y,
+            fee_tier,
+            init_sqrt_price,
+            init_tick,
+            alice
+        )
+        .unwrap();
+
+        approve!(client, TokenRef, token_x, dex, 500, alice).unwrap();
+        approve!(client, TokenRef, token_y, dex, 500, alice).unwrap();
+
+        let pool_key = PoolKey::new(token_x, token_y, fee_tier).unwrap();
+        create_position!(
+            client,
+            InvariantRef,
+            dex,
+            pool_key,
+            -10,
+            10,
+            Liquidity::new(10),
+            SqrtPrice::new(0),
+            SqrtPrice::max_instance(),
+            alice
+        )
+        .unwrap();
+
+        let result = get_liquidity_ticks!(client, InvariantRef, dex, pool_key, 0);
+        assert_eq!(result.len(), 2);
```

Comment:
tbh, it's worth checking the fields of the result array in all cases, not just the length of the result